scripts/Muon/GUI/ElementalAnalysis2/elemental_analysis.py

```diff
@@ -127,6 +150,12 @@ def setup_grouping_changed_observers(self):
         self.grouping_tab_widget.grouping_table_widget.data_changed_notifier.add_subscriber(
             self.auto_tab.auto_tab_presenter.group_change_observer)
 
+        for observer in self.plot_widget.add_or_remove_plot_observers:
```

Comment:
does the data changed observer method not work here? I think it should https://github.com/mantidproject/mantid/pull/31845/files#diff-6b8da9f0a0cd96b092956829f0926b9b859ce09f8ebde6854cfe0e270b49e278R368

Reply:
the data changed observer calls hande data update which does not work well to remove rebinned workpaces well it has been  removed from analysis, while the add_or_remove_plot_observer works  better to remove rebinned workpaces

Reply:
I have just pushed the change I was talking about. Double check it works the way you expect still

Reply:
It works as expected